MouseAimFlight/MouseAimVesselModule.cs

```diff
@@ -173,49 +175,49 @@ void MouseAimPilot(FlightCtrlState s)
                 return;
 
             vesselTransform = vessel.ReferenceTransform;
+            upDirection = VectorUtils.GetUpDirection(vesselTransform.position);
 
             if (s.pitch != s.pitchTrim || s.yaw != s.yawTrim)
             {
-                pitchYawOverrideMouseAim = true;
+                FlyToPosition(s, vesselTransform.up * 5000f + vessel.CoM);
                 return;
             }
             else
-                pitchYawOverrideMouseAim = false;
-
-            upDirection = VectorUtils.GetUpDirection(vesselTransform.position);
-
-            FlyToPosition(s, targetPosition + vessel.CoM);
-        }
-
-        void UpdateMouseCursorForCameraRotation()
-        {
-            if (pitchYawOverrideMouseAim)
             {
-                targetPosition = vesselTransform.up * 5000f;
+                FlyToPosition(s, targetPosition + vessel.CoM);
             }
-            else
-            {
-                Vector3 mouseDelta;
+        }
 
-                if (freeLook)
-                    mouseDelta = Vector3.zero;
-                else
-                    mouseDelta = new Vector3(Input.GetAxisRaw("Mouse X"), Input.GetAxisRaw("Mouse Y")) * MouseAimSettings.MouseSensitivity;
+        void UpdateMouseCursorAndCameraRotation()
+        {
+            Vector3 mouseDelta;
 
-                if (MouseAimSettings.InvertXAxis)
-                    mouseDelta.x *= -1;
-                if (MouseAimSettings.InvertYAxis)
-                    mouseDelta.y *= -1;
+            if (freeLook)
+                mouseDelta = Vector3.zero;
+            else
+                mouseDelta = new Vector3(Input.GetAxisRaw("Mouse X"), Input.GetAxisRaw("Mouse Y")) * MouseAimSettings.MouseSensitivity;
 
-                Transform cameraTransform = FlightCamera.fetch.mainCamera.transform;
+            if (MouseAimSettings.InvertXAxis)
+                mouseDelta.x *= -1;
+            if (MouseAimSettings.InvertYAxis)
+                mouseDelta.y *= -1;
 
-                Vector3d localTarget = cameraTransform.InverseTransformDirection(targetPosition);
-                localTarget += mouseDelta;
-                localTarget.Normalize();
-                localTarget *= 5000f;
+            Transform cameraTransform = FlightCamera.fetch.mainCamera.transform;
 
-                targetPosition = cameraTransform.TransformDirection(localTarget);
+            Vector3d localTarget = cameraTransform.InverseTransformDirection(targetPosition);
+            localTarget += mouseDelta;
+            localTarget.Normalize();
+            if (!freeLook)
```

Comment:
Negating can get out of hand pretty quickly, I would recommend using a "viewLock" or "lockCamera" flag instead.

Reply:
I changed it to look like this: 
if (!freeLook)
    UpdateCameraRotation();
One thing I want to keep is !freeLook, since I simply just want the camera to follow the target when RMB is not pressed...